site/public/js/server.js

```diff
@@ -689,13 +689,171 @@ function downloadCourseMaterialZip(id) {
 }
 
 function checkColorActivated() {
-    var pos = 0;
-    var seq = "&&((%'%'BA\r";
+
+    pos = 0;
+    seq = "&&((%'%'BA\r";
+
+    const magic = () => {
```

Comment:
While I appreciate easter eggs, I am staunchly opposed to this level of obfuscation, and evaling JS code on the fly. It's an attack vector waiting to happen as at a glance it's impossible tell what it's doing, and easy to slip unwarranted things in.